codec-socks/src/main/java/io/netty/handler/codec/socks/SocksAuthResponseDecoder.java

```diff
@@ -41,19 +37,21 @@ protected void decode(ChannelHandlerContext channelHandlerContext, ByteBuf byteB
             throws Exception {
         switch (state()) {
             case CHECK_PROTOCOL_VERSION: {
-                version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
+                SocksSubnegotiationVersion version = SocksSubnegotiationVersion.valueOf(byteBuf.readByte());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)